operator/internal/central/extensions/reconcile_defaulting.go

```diff
@@ -87,24 +87,33 @@ func setDefaultsAndPersist(ctx context.Context, logger logr.Logger, central *pla
 	// This updates central both on the cluster and in memory, which is crucial since this object is used for the final
 	// updating within helm-operator and we have concurrently running controllers (the status controller),
 	// whose changes we must preserve.
-	err := client.Patch(ctx, central, patch)
+	logger.Info("updating defaulting annotations on Central object")
+	err := client.Patch(ctx, u, patch)
 	if err != nil {
 		return errors.Wrap(err, "patching Central annotations")
 	}
-
 	logger.Info("patched Central object",
-		"oldResourceVersion", baseCentral.GetResourceVersion(),
+		"oldResourceVersion", uBase.GetResourceVersion(),
 		"newResourceVersion", central.GetResourceVersion(),
 	)
 
-	// Retain the defaults, which are not in the patched object after cluster refresh.
-	central.Defaults = centralDefaults
+	// If we would not react to a generation mismatch here, this effectively means that the CR spec
+	// currently under reconciliation has changed during the reconciliation flow, specifically during
+	// execution of pre-extensions.
+	// This might not pose a problem given that by our convention the defaulting extension is
+	// expected to run first, but it is cleaner to just abort reconciliation and start over with the new spec.
```

Comment:
Good point. What you're proposing here should also work fine AFAICT, but I have this thought:

I wonder whether rather than special-casing `generation` here (as a proxy for `spec`) and risk missing some corner case we're not thinking about right now, we should do this unconditionally. I.e. this preextension should either:
- have nothing to do and `return nil`
- patch, and return an error such as `Everything is fine, we're just reloading to make sure everything is in sync`.
Then later we can think about replacing this extension with a more tightly integrated mechanism as discussed offline.